File: frontend/src/services/goals-service.ts
import { API_CONFIG } from '@/config/constants';
import type { Goal, GoalFormData } from '@/types';

import { apiClient } from './api-client';
import { BaseService } from './base-service';

class GoalsService extends BaseService<Goal, GoalFormData> {
  constructor() {
    super(API_CONFIG.ENDPOINTS.GOALS);
  }

  async recalculate(id: number): Promise<Goal> {
    return apiClient.post<Goal>(`${this.endpoint}${id}/recalculate/`);
  }

  async reset(id: number): Promise<Goal> {
    return apiClient.post<Goal>(`${this.endpoint}${id}/reset/`);
  }
}

export const goalsService = new GoalsService();

File: apps/mobile/lib/services/goals_service.dart
import '../models/goal.dart';
import 'base_service.dart';

class GoalsService extends BaseService<Goal> {
  GoalsService(super.client)
      : super(
          resourcePath: '/api/v1/personal-planning/goals/',
          fromJson: Goal.fromJson,
          toJson: (goal) => goal.toJson(),
        );
}
